server/form-pages/apply/risks-and-needs/health-needs/mentalHealth.ts

```diff
@@ -82,14 +82,30 @@ export default class MentalHealth implements TaskListPage {
       errors.hasMentalHealthNeeds = 'Confirm whether they have mental health needs'
     }
 
+    if (this.body.hasMentalHealthNeeds === 'yes' && !this.body.needsDetail) {
+      errors.needsDetail = 'Describe mental health needs'
+    }
+
     if (!this.body.isEngagedWithCommunity) {
       errors.isEngagedWithCommunity = 'Confirm whether they are engaged with services'
     }
 
+    if (this.body.isEngagedWithCommunity === 'yes' && !this.body.servicesDetail) {
+      errors.servicesDetail = 'State the services with which they have engaged'
+    }
+
     if (!this.body.hasPrescribedMedication) {
       errors.hasPrescribedMedication = 'Confirm whether they are prescribed medication'
     }
 
+    if (this.body.hasPrescribedMedication === 'yes' && !this.body.isInPossessionOfMeds) {
```

Comment:
as with the previous PR, this and the one below could be nested within a single if statement